File: 240304/한 가지로 열리는 자물쇠/one-way-lock.py
n=int(input())
a,b,c=map(int,input().split())

ans=0
for i in range(1,10):
    for j in range(0,10):
        for k in range(0,10):
            if abs(a-i)<=2 or abs(b-i)<=2 or abs(c-i)<=2:
                ans+=1
print(ans)

File: 260110/한 가지로 열리는 자물쇠/one-way-lock.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const [a, b, c] = input[1].trim().split(' ').map(Number);

// Please Write your code here.

let ans = 0;
for(let i=1; i<=n; i++){
    for(let j=1; j<=n; j++){
        for(let k=1; k<=n; k++){
            if(Math.abs(i-a)<=2 || Math.abs(j-b)<=2 || Math.abs(k-c)<=2){
                ans++;
            }
        }
    }
}

console.log(ans)